crates/holochain_integrity_types/src/entry.rs

```diff
@@ -64,6 +65,17 @@ pub enum Entry {
     CapGrant(CapGrantEntry),
 }
 
+/// The outcome of deserializing [`Entry`] bytes to a user defined type.
+pub enum ParseEntry<T> {
+    /// The bytes successfully deserialized into `T`.
+    Valid(T),
+    /// The bytes failed to parse because of the given reason.
+    Failed(String),
+}
+
+/// The outcome of type checking and parsing a header and entry.
+pub type EntryCheck<T> = ZomeTypeCheck<ParseEntry<T>>;
```

Comment:
How about something like `EntryDef::Found` and `EntryDef::NotInScope`?

Reply:
@jost-s `EntryDef` is already a type.

Reply:
But I'm not happy with the name either

Reply:
@freesig are these types something the zome dev is going to have to interface with?

If so what do you think about collapsing them down to one? Couldn't this just be a Result with the various reasons for not being found or parsed as the Err? Is there a benefit to nesting them?

Reply:
@maackle The idea is to give the user of `deserialize_from_type` more information. So they can tell if it matches the zome id  and entry def index or failed to deserialize or just hit an error calling the host.
You only want to return an error from validation if you want it to try again. Depending on the context you might want to reject if the deserialization fails or zome_id / entry def index mismatches but other times you would just ignore and pass as valid.
However I'm not happy with this PR. Maybe we can think of a better approach?